### ISSUE TITLE
Versioning/Releasing things where the output of a package isn't the package itself

### ISSUE THREAD

**Issue by mitchellhamilton** [atlassian/changesets#171]
This isn't meant to be a let's implement a thing now issue, more a this is just some ideas, we probably won't do anything with for a bit issue.

---

@JedWatson and I were chatting last week about this. 

The first thing I think is separating the parts of Changesets that are the concept of Changesets and the implementation of Changesets.

1. A contributor creates a changeset which declares their intent to release packages with semver bumps and a change summary
2. "Things" are versioned
3. "Things" are released somewhere

The implementation of Changesets adds on some things, most prominently I think they are:

- Bumping dependents
- Publishing to npm

These two things aren't necessary for publishing things where the output of a package isn't the package itself(sites/apps). A problem that we talked about for this as an example is the Keystone site, we have the problem that the site could be ahead of the packages that are published to npm so the information would be wrong for what's actually published - If we versioned the website and only published it when we did releases, we wouldn't have the problem. I think another use case is the style guides - we want to version the style guides and publish the style guide and its parts to npm but I think we could also "publish" from the "source" markdown in packages to the root README and etc.

**Comment by Noviny** [comment]
Clarity thing - this is all hypothetically possible now right? As in, if things are set up as packages within the multi-package repo, they can/will be versioned? And if it depends on packages, updating those packages will cause versioning?

(The atlaskit website gets versioned all the time, but we don't do anything with that fact)

Technically to push changes you need to make your website (or other asset) have a dependency, possibly a fixed version dep, on the package, which could be made easier.

**Comment by mitchellhamilton** [comment]
> this is all hypothetically possible now right? As in, if things are set up as packages within the multi-package repo, they can/will be versioned? And if it depends on packages, updating those packages will cause versioning?

Yes. The main question is how releasing works. This probably won't be a thing in Changesets itself but I thought this would probably be the best place to start a discussion about this.

**Comment by jbolda** [comment]
As a possible extension of this discussion, I am interested in using changesets in a monorepo with multiple languages (in this case, js and rust). Would it be possible to allow the configuration of an arbitrary command for each of the "lifecycles"? I think this potential solution that would cover both my case and the original issue.
